portals/publisher/src/main/webapp/source/src/app/data/APIValidation.js

```diff
@@ -27,12 +27,16 @@ import queryString from 'query-string';
  * @param {string} errorType The joi error type
  * @return {string} simplified error message.
  * */
-function getMessage(errorType, maxLength) {
+function getMessage(errorType, isApiName, maxLength) {
     switch (errorType) {
         case 'any.empty':
             return 'should not be empty';
         case 'string.regex.base':
-            return 'should not contain spaces or special characters';
+            if(isApiName){
+                return 'should not contain trailing or leading spaces, special characters, and consecutive spaces';
+            }else{
```

Comment:
Shall we add spaces here?

Reply:
Fixed in https://github.com/wso2/apim-apps/pull/519/commits/426260ecaebe104033a2bdebe2ec8a03eef3d248